### ISSUE TITLE
digital bookshelf for al-folio

### ISSUE THREAD

**Issue by subhodeeps** [alshedivat/al-folio#923]
Hi,
I would like to request a new feature.

**Is your feature request related to a problem? Please describe.**

This is feature request to add a `digital bookshelf` to al-folio. Many of us, particularly academics and students, read a lot of books and it would be nice if we could display what we are reading on our website and link them to reviews that we may write for them. Think of it as your own personal (stripped-down) Goodreads running on your own personal website.

**Describe the solution you'd like**

I have been recently toying with this idea and created [jekyll-bookworm](https://github.com/subhodeeps/jekyll-bookworm) for the Minima theme. You can see what it looks like [here](https://subhodeeps.github.io/jekyll-bookworm/). Since I am thinking of shifting my academic website to al-folio from  the (the now abandoned) academicpages theme, I decided to see if I can implement it in al-folio. I have been modestly successful. You can take a look at my amatuer tinkering [here](https://subhodeeps.github.io/al-folio/_pages/books/).

I felt other users of al-folio might like this feature and I would be happy to send a pull request (you can, of course, take this idea ahead on your own).

There are however some issues that needs improvement:

- the reviews are not displayed properly on phones and I can't seem to fix it, in general it can do with a lot aesthetic tweaks
- it would be nice to have pagination in the bookshelf but I don't have the required expertise to do it quickly

**Describe alternatives you've considered**

I could not find an alternative or existing solution to what I was looking for: a digital bookshelf and a book review management system. I would be happy if you can point me to something that can implement this in a more polished fashion.

**Additional context**

I am not an expert in web development and I admit my attempts are probably amateurish but I would be happy to get your inputs and see if something like this can be incorporated into al-folio.

Best,
Subhodeep

**Comment by asboyer** [comment]
@subhodeeps I took a look at your personal website and really liked what you've done with your bookshelf. I made an [attempt](https://www.asboyer.com/favs/books/) to do what you were talking about on this al-folio template. I actually prefer the look of yours, as well the different statuses for each book you're reading.

**Comment by george-gca** [comment]
If you manage to implement this, would you mind sending a PR? Maybe you could also add the subscribe via email that I saw you have in your site to this template? It is a feature that have been requested a while ago (#2097).

**Comment by asboyer** [comment]
Yeah, I can do that. I'll make a PR when I've got those.

**Comment by subhodeeps** [comment]
Hi, in the original post I had included my edits to the al-folio theme as well https://github.com/subhodeeps/al-folio and the statuses of the books can be implemented by adding a separate layout for the books and the front-matter https://github.com/subhodeeps/al-folio/blob/master/_layouts/books.html

**Comment by eolesinski** [comment]
@subhodeeps  this is such an incredible idea! I really like how the bookshelf looks on your website. I've been thinking about pulling in your changes here (unless you were waiting to open a PR). The only thing I noticed is that on Safari it seems like the books on your website seem to be very stretched (only showing one per line). Not sure if this is an issue with my computer or if you experienced the same?

**Comment by asboyer** [comment]
You mean stretched like the image is stretched and out of proportion or stretched like each book is on one line? If its the latter, that's done on purpose. I like having the list instead of a grid view, that was something I've previously done on [older websites](https://asboyer.pythonanywhere.com/books/library).

**Comment by george-gca** [comment]
Can you submit a PR with this? I've seen a few users asking for it.

**Comment by subhodeeps** [comment]
Hi, thanks for the interest in this idea! I have no problem if someone else goes ahead and incorporates the code into al-folio. My attempts to incorporate this into al-folio can be found here https://subhodeeps.github.io/al-folio/_pages/books/

I am bit occupied at the moment to submit a PR request and it has been a while since I looked at al-folio's code, so it'd be difficult for me to submit a PR right now. I would be delighted if somebody took takes this idea forward!

**Comment by asboyer** [comment]
> Can you submit a PR with this? I've seen a few users asking for it.

Yea, I'll try to work on it. Thanks for the help @subhodeeps

**Comment by subhodeeps** [comment]
Hi, the changes I made to al-folio is available in my fork of the al-folio repo https://github.com/subhodeeps/al-folio

The commits are available here (6th Oct): https://github.com/alshedivat/al-folio/compare/master...subhodeeps:al-folio:master

Since I have made several other changes to the site, I can't (shouldn't) create a pull request but I hope the commits will give you an idea of the changes that needs to be made.

**Comment by george-gca** [comment]
I started implementing a mix-up version of both your solutions. I chose @subhodeeps version to list the books and @asboyer version to display the review itself. There are still some fixing to do though. Someday I can come to this again to finish its implementation.

**Comment by asboyer** [comment]
@george-gca awesome. Gotten super busy with schoolwork but I can also take a look at it this weekend if you want help.

**Comment by george-gca** [comment]
@asboyer @subhodeeps can you please check my new PR #2866? You'll have to test it locally, since for some reason the build on Netlify is failing.

**Comment by asboyer** [comment]
Okay, I am going to take a look at this now, sorry for the late response!

**Comment by subhodeeps** [comment]
> @subhodeeps this is such an incredible idea! I really like how the bookshelf looks on your website. I've been thinking about pulling in your changes here (unless you were waiting to open a PR). The only thing I noticed is that on Safari it seems like the books on your website are very stretched (only showing one per line). Not sure if this is an issue with my computer or if you experienced the same?

Hi, I fixed this issue. Take a look https://github.com/subhodeeps/subhodeeps.github.io/blob/master/_layouts/books.html

**Comment by subhodeeps** [comment]
> #2866

This looks good! 

https://deploy-preview-2866--relaxed-lollipop-b6bc17.netlify.app/books/

**Comment by asboyer** [comment]
> https://github.com/subhodeeps/subhodeeps.github.io/blob/master/_layouts/books.html

It looks really good, yeah. I like he combined the best elements of each of our implementations. @george-gca is there anything else you need help with?

**Comment by george-gca** [comment]
Yes, there is a small layout issue when visualizing the book review in mobile layout. I believe that's the only thing missing, but I haven't had the time to look back at it. Can you guys take a look at this @asboyer @subhodeeps?
